From 07d860fab2284e26f394d22e59b68ee3973ac366 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Fri, 21 Jan 2022 23:27:24 +0100
Subject: [PATCH 322/456] arm64: dts: rk3399-pinephone-pro: Add support for
 Pinephone keyboard

Pinephone keyboard can be used with Pinephone Pro. Describe it in
the device tree.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index a8d04b69944c..02a0c4b02187 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -961,6 +961,15 @@ &i2c5 {
 	i2c-scl-falling-time-ns = <15>;
 	pinctrl-0 = <&i2c5_xfer &pogo_int_pin>;
 	status = "okay";
+
+	keyboard@15 {
+		compatible = "pine64,kb151";
+		reg = <0x15>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <RK_PA0 IRQ_TYPE_EDGE_FALLING>;
+		vbat-supply = <&vcc5v0_sys>;
+		wakeup-source;
+	};
 };
 
 &i2s0 {
-- 
2.34.1

